c++/test/tfile.cpp

```
@@ -80,7 +80,7 @@ test_file_create()
     H5File *file1 = NULL;
     try {
         // Create file FILE1
-        file1 = new H5File(FILE1, H5F_ACC_EXCL);
+        file1 = new H5File(FILE1, H5F_ACC_TRUNC);
```

Why the change in file access mode here?

---

The test is creating a new file. H5F_ACC_EXCL causes it to fail if the file already exists, which is what was happening. It was failing silently before, but with the 1.8 format it causes issues due to a fapl mismatch - see #5954

---

I see. Fine either way, though previously the issues with this test involved the fact that it left a file open unintentionally and also forgot to delete it, so I wonder if that isn't the case here as well.

---

I think this is at least partly due to it simply using HDremove() to remove the test files, which obviously doesn't work with the family file driver. Something that should be fixed for sure, but not within the scope of this PR